warehouse/query-core/src/main/java/datawave/query/postprocessing/tf/TermOffsetMap.java

```diff
@@ -0,0 +1,115 @@
+package datawave.query.postprocessing.tf;
+
+import datawave.query.jexl.functions.TermFrequencyList;
+import org.javatuples.Pair;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.StringJoiner;
+
+/**
+ * This encapsulates term frequencies information that is used when evaluating content:phrase and content:within functions in a JEXL query. It is also used to
+ * store phrase indexes that have been found for matching hits so that we can retrieve excerpts when requested.
+ */
+public class TermOffsetMap {
```

Comment:
Does this class support merging with another `TermOffsetMap`?

Reply:
Currently no, there was code in TermOffsetPopulator that merged two term frequency lists, but Ivan verified that it was not in use anywhere, so I deleted it and didn't implement support for it for TermOffsetMap. If need be, I can implement it.

Reply:
Let's add merging TermOffsetMaps as an issue to be done at some point in the future. There's currently no use case, but at some point we may want to shard documents which would require merging them.

Reply:
Sounds good to me.